plutus-core-spec/plutus-core-specification.tex

```diff
@@ -828,7 +830,10 @@ \subsection{Term reduction}
 Execution of Plutus Core programs is performed by (possibly
 non-terminating) reduction of well-typed terms.  The reduction rules
 are contained in Figure~\ref{fig:term-reduction}, and give us a fairly
-standard operational semantics.  
+standard operational semantics.  We use $[V/x]M$ to denote
+substitution of a value for a variable in a term, and $\{A/\alpha\}V$
+to denote instantation of a polymorphic value (or term) $V$ at a
+particular type $A$. All substitutions are capture-avoiding, as usual.
```

Comment:
I think it's fairly normal to just use the same substitution operator for both. if anything, the curly bracket notation is a bit confusing since it looks like our syntax.

Reply:
Also, in the previous diff you say you are going to use the normal substitution syntax for type substitutions!

(As usual, I suggest using a macro so these sorts of changes are easier...)

Reply:
Yeah, I swithered over that for a while and then thought I'd use a different syntax for this because we use different syntax for type instantiation vis-a-vis application of small lambdas at the type and term levels.  I don't really mind though.

[Yes, I did say I was going to use the normal syntax for type substitutions, but this is somehow a bit different.  But meh.]

Reply:
Well, we certainly shouldn't use different syntax for type substitutions in two places! I suggest just using the normal one.

Reply:
We have three kinds of application and three kinds of substitution. I think it makes perfect sense to try to match them up visually. However, we also use curly brackets for frames and as Michael says I guess it is more conventional to use square brackets for substitution. We are paying the price for spending our brackets on application. I vote for using square brackets for all three kinds of substitution.

Reply:
OK, I've fixed the substitution syntax now.